engine/runtime/src/main/java/org/enso/interpreter/runtime/library/dispatch/TypeOfNode.java

```diff
@@ -309,4 +397,17 @@ boolean isNone() {
       }
     }
   }
+
+  static final class NonTypeResult extends Exception {
+    final Object result;
+
+    NonTypeResult(Object result) {
+      this.result = result;
+    }
+
+    @Override
+    public Throwable fillInStackTrace() {
```

Comment:
Overriding `fillInStackTrace()` to no-op should make _throwing on this exception_ as fast as `GOTO`.

Reply:
However, let's check the benchmarks are fine - [engine ones](https://github.com/enso-org/enso/actions/runs/11957075788) and [stdlib ones](https://github.com/enso-org/enso/actions/runs/11957100119)

Reply:
Benchmarks aren't affected by this change.